### ISSUE TITLE
[BUG] Crash when loading some FreeDOS code page files

### ISSUE THREAD

**Issue by FeralChild64** [dosbox-staging/dosbox-staging#1506]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

v0.79.0-alpha-48-ge190f775

### What Operating System are you using?

Linux x86_64

### If Other OS, please describe

Gentoo Linux, gcc version 11.2.0 (Gentoo 11.2.0 p1) 

### Relevant hardware info

AMD64

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

The built-in KEYB.COM sometimes crashes (takes the whole DOSBox down) when trying to load compressed code page from external file. Crash seems to be random, sometimes it happens just about every time, sometimes I have to be patient. I couldn't reproduce the problem using DOSBox X 0.83.20. It happens only with UPX compressed code pages (as provided by FreeDOS); when I first decompressed the code page using `upx -d <file>`, it never crashed so far. 

### Steps to reproduce the behaviour.

1. Get EGA10.CPX from https://gitlab.com/FreeDOS/base/cpidos/-/tree/master/CPI
2. MOUNT C <directory with EGA10.CPX>
3. C:
4. KEYB PL 668 EGA10.CPX
5. crashes with SIGSEGV

### Your configuration

```ini
default - I have deleted dosbox-staging.conf to be sure
```

### Provide a Log

```text
https://gist.github.com/FeralChild64/6a5e8482f869e9a4a48f33bf0f01fabf
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by LowLevelMahn** [comment]
the FreeDOS program mode can CPX files:
https://github.com/FDOS/mode/blob/fb3e8c32b49334a1a6add05f8b44f784c3f41d66/cpitrick.asm#L26
(the asm source is super trivial - so i think easily portable)

I can't find any CPX/UPX reference in the 10 years old KEYB source of FreeDOS
https://github.com/FDOS/keyb/blob/master/SOURCE/KEYB.PAS
is that the tool that works with these CPX files?

**Comment by Wengier** [comment]
I had confirmed it is a regression in the 0.78.0 release. The details are shared in Discord.

**Comment by Grounded0** [comment]
Fixed by #1507.
